scipy/_lib/_util.py

```diff
@@ -480,3 +482,98 @@ def rng_integers(gen, low, high=None, size=None, dtype='int64',
 
         # exclusive
         return gen.randint(low, high=high, size=size, dtype=dtype)
+
+
+def _argmin(a, keepdims=False, axis=None):
+    """
+    argmin with a `keepdims` parameter.
+
+    See https://github.com/numpy/numpy/issues/8710
+
+    If axis is not None, a.shape[axis] must be greater than 0.
+    """
+    res = np.argmin(a, axis=axis)
+    if keepdims and axis is not None:
+        res = np.expand_dims(res, axis=axis)
+    return res
+
+
+def _first_nonnan(a, axis):
```

Comment:
Not super familiar with `_lib` setup, do we ***need*** to keep the function names "private" when the sub-package itself is already private?

Reply:
This is just "an overabundance of caution".  Suppose this function is imported and used in, say, `stats.py`.  Then someone could do `from scipy.stats.stats import _first_nonnan`.  Without the underscore prefix, it might then appear to be a public function.  Maybe that's a stretch, but that's why I refer to using the underscore as an overabundance of caution.

Reply:
ok, thanks for clarifying, let's leave as is, better safe than sorry.